app/containers/CreateProjectDialog/CreateProjectDialog.js

```diff
@@ -101,38 +105,108 @@ class CreateProjectDialog extends Component {
     // Progression for Clone Directory
     {
       step: 'CloneProjectDetails',
-      next: 'SelectAssets',
+      next: 'Create', 
       prev: 'SelectProjectType',
     },
-    {
-      step: 'SelectAssets',
-      next: 'Create',
-      prev: 'CloneProjectDetails',
-    },
   ];
 
-  handleSelectAddProject(type) {
+  handleSourceDirectoryChanged(dir) {
+    const { isValid, errorMessage } = this.validateCloneDirectories(
+      dir,
+      this.state.project.targetBaseDirectory,
+      this.state.project.name
+    );
+  
     this.setState((prevState) => ({
       project: {
         ...prevState.project,
-        type: type,
+        sourceDirectory: dir,
       },
+      canProgress: isValid,
+      errorMessage: errorMessage
     }));
-    switch (type) {
-      case Constants.ProjectType.NEW_PROJECT_TYPE:
-        this.setState({ step: 'SelectNewProjectTemplate' });
-        break;
-      case Constants.ProjectType.EXISTING_PROJECT_TYPE:
-        this.setState({ step: 'ExistingProjectDetails' });
-        break;
-      case Constants.ProjectType.CLONE_PROJECT_TYPE:
-        this.setState({ step: 'CloneProjectDetails' });
-        break;
-      default:
-        this.setState({ step: 'SelectProjectType' });
+  }
+  
+  handleTargetBaseDirectoryChanged(dir) {
+    const { isValid, errorMessage } = this.validateCloneDirectories(
+      this.state.project.sourceDirectory,
+      dir,
+      this.state.project.name
+    );
+  
+    this.setState((prevState) => ({
+      project: {
+        ...prevState.project,
+        targetBaseDirectory: dir,
+      },
+      canProgress: isValid,
+      errorMessage: errorMessage
+    }));
+  }
+  
+  validateCloneDirectories(sourceDir, targetBaseDir, name) {
+    const hasRequiredFields = sourceDir && sourceDir !== '' && 
+           targetBaseDir && targetBaseDir !== '' && 
+           name && name !== '';
+    
+    if (!hasRequiredFields) return { isValid: false, errorMessage: null };
+    // Check if user is trying to clone a directory onto itself
+    if (sourceDir === targetBaseDir) {
```

Comment:
The validation checks are great!  I think there's just one more scenario to handle:
```
sourceDir = '/my/path/dir1'
targetBaseDir = '/my/path'
name = 'dir1'
```
Check here that `targetBaseDir` joined with `name` is not the same path as `sourceDir`.